core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2342,6 +2342,12 @@ private[spark] object Utils extends Logging {
    * Return the initial number of executors for dynamic allocation.
    */
   def getDynamicAllocationInitialExecutors(conf: SparkConf): Int = {
+    if (conf.get(DYN_ALLOCATION_INITIAL_EXECUTORS) < conf.get(DYN_ALLOCATION_MIN_EXECUTORS)) {
```

Comment:
this isn't going to warn if executor instances was specified and is less then the minimum.  We could simply move this check til after the max and see if the number that is going to be returned is < min executors.